cmd/controller/diskstore.go

```diff
@@ -57,6 +57,18 @@ func (c *Controller) CreateDisk(dr apis.Disk) {
 	}
 }
 
+// UpdateDiskCR to edit the DiskCR in etcd
+func (c *Controller) UpdateDiskCR(newDr *apis.Disk) error {
```

Comment:
This function and UpdateDisk can be folded in to one. As we discussed how to fix it, try to make it that way.

Reply:
We have completely got rid of the `UpdateDiskCR()` . The same functionality will now be done by `UpdateDisk()`